### DISCUSSION TITLE
Dotfiles for different systems or what

### DISCUSSION THREAD

**Discussion by Ra2-IFV** [twpayne/chezmoi#3964] [Q&A] [answered]
I read [this](https://www.chezmoi.io/user-guide/frequently-asked-questions/design/#can-i-use-chezmoi-to-manage-files-outside-my-home-directory), so `chezmoi` is not suitable for managing files outside home directory, even I'm the only user in system 😢 going to learn ansible
But I still have some questions, how to apply specific dotfiles only when conditions are met? I use `chezmoi` on multiple systems (windows/unix), so I have system-specific dotfiles.
I also see this [include-files-from-elsewhere](https://www.chezmoi.io/user-guide/include-files-from-elsewhere/), is it recommended to split the only `chezmoi` repo into more repos? For example, `dotfiles`, `dotfiles-windows` and `dotfiles-linux`, manage them by editing `.chezmoiexternal.toml`

**Comment by halostatue** [comment]
I believe you have misapprehended the purpose, scope, and power of chezmoi.

As stated, chezmoi is a home directory ("dotfiles") manager. That does not mean that it cannot interact with or modify system state, but that doing so requires scripts and accepted programmatic interfaces for modifying such state. For example, on macOS, you can use the `defaults` program to change system-wide settings and `brew` or `macports` or other such mechanisms to install tools on your system. On windows, you could use scripts to call regedit and make registry changes. On Linux, I don't know as I don't really use it (though #3659 points to an interesting option and way of using chezmoi).

With respect to multiple platform support, this is *natively* supported by templating and scripting, and there are extensive examples in the documentation, in the show-and-tell section of the discussions, and linked to in various issues. I have *some* consideration for Linux in my dotfiles although I am exclusively on macOS ([halostatue/dotfiles](https://github.com/halostatue/dotfiles)). Others to look at are [twpayne/dotfiles](https://github.com/twpayne/dotfiles), [felipecrs/dotfiles](https://github.com/felipecrs/dotfiles), and [bradenhilton/dotfiles](https://github.com/bradenhilton/dotfiles). Most of these and more can be found with the [chezmoi topic search](https://github.com/topics/chezmoi).

**Reply by Ra2-IFV** [reply]
I know, I'm not asking `chezmoi` to manage files outside home directory, I'm seeking for different home directory setups for different systems. I mean, if I'm running linux, `chezmoi` will generate some configs that only for other systems (scoop, pwsh, brew etc.) It's fine to keep them there but not graceful enough, and I found this in examples:
`.chezmoiexternal.toml.tmpl`
```
{{- if eq .chezmoi.os "linux" }}
[".config/xxx"]
  type = "xxx"
  ...
{{- end }}
```
That's what I'm looking for I guess? Store them in different repos, then checkout repos with template functions.
I know chezmoi is very powerful, but I'm also a bit confused after I read all the docs, I'm dumb. The examples are very useful for me, and I have never tried to check for it. Thanks.

**Reply by halostatue** [reply, marked as answer]
Look at [.chezmoiignore](https://www.chezmoi.io/reference/special-files-and-directories/chezmoiignore/). Externals should be used only for full externals (like oh-my-zsh, if that's your thing).

**Reply by bradenhilton** [reply]
@Ra2-IFV You don't need separate repos or branches. You can use [`.chezmoiignore`](https://www.chezmoi.io/reference/special-files-and-directories/chezmoiignore/) to prevent these files from appearing on systems they aren't supposed to be on.

[Manage machine-to-machine differences](https://www.chezmoi.io/user-guide/manage-machine-to-machine-differences) also outlines a few methods.

**Reply by Ra2-IFV** [reply]
> Look at [.chezmoiignore](https://www.chezmoi.io/reference/special-files-and-directories/chezmoiignore/). Externals should be used only for full externals (like oh-my-zsh, if that's your thing).

Oh fuck that's great, I misunderstood this because I thought it works like `.gitignore`, applies when running `chezmoi add`
https://www.chezmoi.io/user-guide/manage-machine-to-machine-differences/#ignore-files-or-a-directory-on-different-machines
Just be more careful when reading docs

**Reply by halostatue** [reply]
Do note that `.chezmoiignore` does *not* work like `.gitignore` in sometimes surprising ways. In particular, recursive matches (`**`) and negation (`!`) sometimes interact in surprising (but entirely understandable) ways. The best way to check this is with `chezmoi unmanaged` and `chezmoi ignored` (this shows things that are managed, but are incorrectly excluded by `.chezmoiignore`).

See my [`.chezmoiignore`](https://github.com/halostatue/dotfiles/blob/main/home/.chezmoiignore) for details (it will be updated shortly with some tweaks based on `chezmoi ignored`).
